### PR TITLE
Skip adding alias targets when they don't exist

### PR DESCRIPTION
**Pull request by porglezomp**
The CMake uses the concept of a "primary variant" which isn't necessarily aligned with either the host or target. In some cases, like cross-compiling an iOS compiler toolchain for `macosx-arm64`, the expected "primary variant" target will be missing and so CMake will fail. We can skip adding the alias since `build-script` will call the more specific target anyway.

### CODE REVIEW

**Comment by porglezomp** [comment]
@swift-ci please smoke test

File: stdlib/CMakeLists.txt [hunk 1]
```diff
@@ -196,13 +196,23 @@ function(swift_create_stdlib_targets name variant define_all_alias)
...
-  set_target_properties(${name}${variant}
-    PROPERTIES
-    FOLDER "Swift libraries/Aggregate")
+
+  # When cross-compiling host tools for multiple architectures, targeting a
+  # different SDK, the primary variant is not one of the variants being built,
+  # so it can't be added as a target here. build-script will invoke the
+  # more-specific target, so just skip creating this target and warn in case
+  # someone is using the CMake more directly.
+  if(TARGET ${name}${SWIFT_PRIMARY_VARIANT_SUFFIX}${variant})
```
Comment (edymtt) [inline stdlib/CMakeLists.txt:205]: I believe a more readable check would be

```suggestion
  if(SWIFT_PRIMARY_VARIANT IN_LIST SWIFT_SDKS)
```

File: stdlib/CMakeLists.txt [hunk 2]
```diff
@@ -196,13 +196,23 @@ function(swift_create_stdlib_targets name variant define_all_alias)
...
+    add_custom_target(${name}${variant}
+      ${ALL_keyword}
+      DEPENDS
+      ${name}${SWIFT_PRIMARY_VARIANT_SUFFIX}${variant})
+    set_target_properties(${name}${variant}
+      PROPERTIES
+      FOLDER "Swift libraries/Aggregate")
+  else()
+    message(WARNING "Primary variant is not being built, target ${name}${SWIFT_PRIMARY_VARIANT_SUFFIX}${variant} does not exist, not creating ${name}${variant} alias target.")
```
Comment (edymtt) [inline stdlib/CMakeLists.txt:214]: If we decide to change the check, we could improve on this message as well, e.g.

```suggestion
    message(WARNING "Primary variant ${SWIFT_PRIMARY_VARIANT} is not being built, not creating ${name}${variant} alias target for it.")
```

Comment (edymtt) [reply stdlib/CMakeLists.txt:214]: (Optional) would it be possible to add some tests for this scenario?

**Review by edymtt — CHANGES_REQUESTED** [review]

**Review by edymtt — APPROVED** [review]

**Comment by porglezomp** [comment]
@swift-ci please smoke test

**Comment by porglezomp** [comment]
I'm also rebuilding locally against a preset that I'm sure exercises cross-compilation

**Comment by porglezomp** [comment]
@swift-ci please smoke test macOS Platform

**Comment by porglezomp** [comment]
@swift-ci please smoke test macOS Platform

File: stdlib/CMakeLists.txt [hunk 3]
```diff
@@ -196,13 +196,23 @@ function(swift_create_stdlib_targets name variant define_all_alias)
...
-  set_target_properties(${name}${variant}
-    PROPERTIES
-    FOLDER "Swift libraries/Aggregate")
+
+  # When cross-compiling host tools for multiple architectures, targeting a
+  # different SDK, the primary variant is not one of the variants being built,
+  # so it can't be added as a target here. build-script will invoke the
+  # more-specific target, so just skip creating this target and warn in case
+  # someone is using the CMake more directly.
+  if(SWIFT_PRIMARY_VARIANT IN_LIST SWIFT_SDKS)
```
Comment (edymtt) [inline stdlib/CMakeLists.txt:205]: I fat fingered a variable name in my previous suggestion, this should be the correct one

```suggestion
  if(SWIFT_PRIMARY_VARIANT_SDK IN_LIST SWIFT_SDKS)
```

File: stdlib/CMakeLists.txt [hunk 4]
```diff
@@ -196,13 +196,23 @@ function(swift_create_stdlib_targets name variant define_all_alias)
...
+    add_custom_target(${name}${variant}
+      ${ALL_keyword}
+      DEPENDS
+      ${name}${SWIFT_PRIMARY_VARIANT_SUFFIX}${variant})
+    set_target_properties(${name}${variant}
+      PROPERTIES
+      FOLDER "Swift libraries/Aggregate")
+  else()
+    message(WARNING "Primary variant ${SWIFT_PRIMARY_VARIANT} is not being built, not creating ${name}${variant} alias target for it.")
```
Comment (edymtt) [inline stdlib/CMakeLists.txt:214]:
```suggestion
    message(WARNING "Primary variant ${SWIFT_PRIMARY_VARIANT_SDK} is not being built, not creating ${name}${variant} alias target for it.")
```

**Review by edymtt — CHANGES_REQUESTED** [review]

**Review by compnerd — APPROVED** [review]
I believe that this should be safe.

**Review by edymtt — APPROVED** [review]

**Comment by porglezomp** [comment]
Just squashing the two review fixup commits and improving their commit message.

**Comment by porglezomp** [comment]
@swift-ci please smoke test and merge

**Comment by porglezomp** [comment]
@swift-ci please smoke test and merge

**Comment by porglezomp** [comment]
@swift-ci please smoke test macOS Platform and merge

**Comment by porglezomp** [comment]
@swift-ci please smoke test macOS Platform

**Comment by porglezomp** [comment]
@swift-ci please smoke test macOS Platform

**Pull request merged**